### PR TITLE
thermostat schedules

### PR DESCRIPTION
**Pull request by LnL7**

### CODE REVIEW

File: src/gateway/pubsub.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 if False:  # MYPY
     from typing import Callable, Dict, List, Literal, Tuple, Union
-    GATEWAY_TOPIC = Literal['config', 'state']
+    GATEWAY_TOPIC = Literal['config', 'scheduler', 'state']
```
Comment (svanscho) [inline src/gateway/pubsub.py:31]: would change this to INTENT or ACTION

**Review by svanscho — CHANGES_REQUESTED** [review]

File: src/gateway/scheduling_controller.py [hunk 2]
```diff
@@ -94,71 +100,64 @@ def set_webinterface(self, web_interface):
...
+            self._abort(schedule_dto)
+            self._schedules.pop(schedule_dto.id, None)
+
+    def _submit(self, schedule_dto):
         # type: (ScheduleDTO) -> None
-        job_id = str(schedule_dto.id)
         if schedule_dto.status == 'ACTIVE':
-            job = self._jobs.get(job_id)
+            logger.debug('Submitting schedule %s', schedule_dto)
+            job_id = 'schedule.{0}'.format(schedule_dto.id)
```
Comment (LnL7) [inline src/gateway/scheduling_controller.py:136]: just dto job_id

File: src/gateway/scheduling_controller.py [hunk 3]
```diff
@@ -169,29 +168,74 @@ def load_schedule(self, schedule_id):
 
     def load_schedules(self, source=Schedule.Sources.GATEWAY):
```
Comment (LnL7) [inline src/gateway/scheduling_controller.py:169]: cleanup

File: src/gateway/scheduling_controller.py [hunk 4]
```diff
@@ -219,6 +260,22 @@ def _execute_schedule(self, schedule_dto):
...
 
+    def _update_status(self, schedule_dto):
+        # type: (ScheduleDTO) -> None
+        if schedule_dto.has_ended:
+            schedule_dto.next_execution = None
+            schedule_dto.status = 'COMPLETED'
+            schedule = ScheduleMapper.dto_to_orm(schedule_dto)
+            schedule.save()
+        else:
+            job_id = 'schedule.{0}'.format(schedule_dto.id)
```
Comment (LnL7) [inline src/gateway/scheduling_controller.py:271]: use dto job_id

**Review by svanscho — APPROVED** [review]

**Pull request merged**